primitives/src/signatures/schnorr.rs

```diff
@@ -50,32 +51,34 @@ where
     /// Signature
     type Signature = Signature<P>;
 
-    /// message
-    type MessageUnit = P::BaseField;
+    /// A message is &\[MessageUnit\]
+    type MessageUnit = F;
 
     /// generate public parameters from RNG.
-    fn setup<R: CryptoRng + RngCore>(
+    fn param_gen<R: CryptoRng + RngCore>(
```

Comment:
Q: In what cases do we need `PublicParameter`? Does a group generator count as part of pp?

Reply:
good question. I was thinking to pub the group generators there -- this allows users to use customized generators other than the one in Arkworks. I haven't thought about the security implications though.

Reply:
hmmm in https://github.com/EspressoSystems/jellyfish/pull/55/commits/19d9c81b42ca8481d8d94c4fcb8f5b4d497e588f in wrote a potential solution. The api is a bit strange though, when the `default generator` is called.